torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -172,6 +172,98 @@ std::unordered_map<Val*, Val*> getSimplificationMap(Fusion* fusion) {
   return extent_to_min_input_id_extent;
 }
 
+bool shouldFuseBroadCast(const BroadcastOp* broadcast) {
```

Comment:
It seems like the check here should be:
The broadcast and reduction operation are in the same for loop (excluding anything with size-1 dimension)
The reduction is only significant across TIDx (excluding anything with size-1 dimension)
The broadcast doesn't broadcast across any other block dimension other than TIDx

I'd like this to either more closely match how we generate for loops, or actually inspect the for loops. The other two checks are straight forward.

I think this would be an okay first implementation, I just don't know how much I trust it longer term.

Reply:
I see. Thanks for the suggestion. Based on the previous discussions probably both the checking of loop nest and the actual "fusing" of the broadcast part are easier on kernel ir. I will try to do a quick experiment this week and maybe move these into kernel IR in this PR if that's straightforward enough.